### ISSUE TITLE
No more entities available after HA Core Update 2024.1.0

### ISSUE THREAD

**Issue by silub80** [PlusPlus-ua/ha_mobilealerts#23]
Hi, 
after Update to HA Core 2024.1.0 no more mobile alerts enties are available. 
I´m running HA OS 11.2 in a Virual Machine on a Synology DS920+

I´m using TFA30.3060.01 & MA10120.

Maybe anyone could help to solve the issue?

Many Thanks

**Comment by kalleylimaki** [comment]
I have the same problem.
My sensor log alert.

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 360, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 234, in async_setup_entry
    async_add_entities(create_gateway_binary_sensor_entities(coordinator.gateway))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 190, in create_gateway_binary_sensor_entities
    return [
           ^
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 191, in <listcomp>
    MobileAlertesGatewayBinarySensor(
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 79, in __init__
    description.translation_key = description.key
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __setattr__
dataclasses.FrozenInstanceError: cannot assign to field 'translation_key'

**Comment by ilwalti** [comment]
Ditto.
I think that this project was abandoned... too bad.

**Comment by AfRotaker** [comment]
Is there some way to contact the author?

**Comment by ilwalti** [comment]
Don't know... he (@PlusPlus-ua) is silent since 6 months ago.

**Comment by dcear** [comment]
Hello GuysThe developper live... In UA 🇺🇦..... I hope he and his family are well. Sincerely

**Comment by RaRo1980** [comment]
at the moment i changed to this integration and works well
https://github.com/CestLaGalere/mobilealerts

**Comment by silub80** [comment]
CestLaGalere works but cloud based only. It's better than nothing.

The Advantage from ha_mobilealerts is 100% local with much more faster update intervals. Hopefully it will work again soon

**Comment by newslettersns2000** [comment]
I have the same problem.

**Comment by gianb80** [comment]
Me too :-(

**Comment by rodtrip2** [comment]
The problem seems to be with a missing python library :

`2024-01-08 12:34:30.497 ERROR (MainThread) [homeassistant.config_entries] Error importing platform config_flow from integration mobile_alerts to set up mobile_alerts configuration entry: cannot import name 'CONF_GATEWAY' from 'custom_components.mobile_alerts.const' (/config/custom_components/mobile_alerts/const.py)`

A lot of custom integrations seem to have similar problems. Hope that home assistant publishes a global fix, or that somebody knows enough about python to fix this addon.

**Comment by larobo** [comment]
The problem seems to be that "suddenly" Home assistant core made the EntityDescription object immutable. A good description and and example fix here: https://github.com/ben8p/home-assistant-bunq-balance-sensors/pull/15.

After correcting the installed Python files (binary_sensor.py and sensor.py) Mobile-Alerts seems to work again with HA 2024.1.2.

**Comment by rodtrip2** [comment]
> The problem seems to be that "suddenly" Home assistant core made the EntityDescription object immutable. A good description and and example fix here: [ben8p/home-assistant-bunq-balance-sensors#15](https://github.com/ben8p/home-assistant-bunq-balance-sensors/pull/15).
> 
> After correcting the installed Python files (binary_sensor.py and sensor.py) Mobile-Alerts seems to work again with HA 2024.1.2.

Thanks for finding the problem! Would it be possible to tell us what you changed exactly, or to fork the custom integration?

**Comment by larobo** [comment]
I would expect HA to roll-back the pre-mature immutable "feature". I've attached the fixes I've made (done without much insight in HA development). They work for me but use at your own risk,

The fixes are simple, like replace
```
description.translation_key = description.key
```
with this
```
description = dataclasses.replace(
    description,
    translation_key = description.key
)
```

[2024.1-fixes.zip](https://github.com/PlusPlus-ua/ha_mobilealerts/files/13870556/2024.1-fixes.zip)

**Comment by gianb80** [comment]
Many thanks, but even if HA releases the roll-back, then this FIX will still be fine, since in the future HA will bring this implementation back into effect, correct?

**Comment by Iminet72** [comment]
Thanks for the fix
I put it together and asked for combing, if it doesn't work, I'll offer the fork I have for use

**Comment by Iminet72** [comment]
I've been using my own fork for a long time now.
I try to keep it fresh. although I'm just now making friends with programming.
If anyone wants to use it, it works for me

**Comment by AfRotaker** [comment]
> I would expect HA to roll-back the pre-mature immutable "feature". I've attached the fixes I've made (done without much insight in HA development). They work for me but use at your own risk,
> 
> The fixes are simple, like replace
> 
> ```
> description.translation_key = description.key
> ```
> 
> with this
> 
> ```
> description = dataclasses.replace(
>     description,
>     translation_key = description.key
> )
> ```
> 
> [2024.1-fixes.zip](https://github.com/PlusPlus-ua/ha_mobilealerts/files/13870556/2024.1-fixes.zip)

Thank you so much!

**Comment by yah996** [comment]
Thanks for your update...

Found another problem. 2024.x HA can not publish (MQTT) the state attribute "last_update" anymore. So bad.

MQTT Publish code
payload:   {{ state_attr('sensor.temperature_sensor_in_cable_0XXXXXXXXXXX_cable_temperature','last_update')    }}
 
{{  ( state_attr('sensor.temperature_sensor_in_cable_0XXXXXXXXXX_cable_temperature','last_update')) }} says None!

Thanks for your help.

**Comment by yah996** [comment]
Fixed it... I changed last_update to last_updated

**Comment by silub80** [comment]
> Fixed it... I changed last_update to last_updated

where did you fixed that - in binary_sensor.py and sensor.py only?

**Comment by yah996** [comment]
I changed my automation, I left the py script files as there were. It works for me.

**Comment by silub80** [comment]
I think we can mark this issue as solved trough [larobo](https://github.com/larobo)

https://github.com/PlusPlus-ua/ha_mobilealerts/issues/23#issuecomment-1882555904

**Comment by hiatsu0** [comment]
I suspect I might be stupid somehow, but the change by @larobo does not work for me somehow. I get the following errors instead of the [original ones](https://github.com/PlusPlus-ua/ha_mobilealerts/issues/23#issuecomment-1877555414):


```
Logger: homeassistant
Source: custom_components/mobile_alerts/binary_sensor.py:181
Integration: Mobile-Alerts
First occurred: 18.30.09 (5 occurrences)
Last logged: 18.35.22

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 243, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 399, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in async_update_listeners
    update_callback()
  File "/config/custom_components/mobile_alerts/base.py", line 204, in _handle_coordinator_update
    self.update_data_from_sensor()
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 181, in update_data_from_sensor
    (int(time_span_sensor._attr_native_value) == 0) and
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'unavailable'
```

and

```
Logger: aiohttp.server
Source: /usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py:421
First occurred: 18.32.42 (2 occurrences)
Last logged: 18.34.22

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/mobilealerts/proxy.py", line 87, in request_handler
    await gateway.handle_update(
  File "/usr/local/lib/python3.11/site-packages/mobilealerts/gateway.py", line 526, in handle_update
    await self.handle_sensors_update(packages)
  File "/usr/local/lib/python3.11/site-packages/mobilealerts/gateway.py", line 510, in handle_sensors_update
    await self.handle_sensor_update(
  File "/usr/local/lib/python3.11/site-packages/mobilealerts/gateway.py", line 496, in handle_sensor_update
    await self._handler.sensor_updated(sensor)
  File "/config/custom_components/mobile_alerts/coordinator.py", line 52, in sensor_updated
    self.async_set_updated_data({})
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 426, in async_set_updated_data
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in async_update_listeners
    update_callback()
  File "/config/custom_components/mobile_alerts/base.py", line 204, in _handle_coordinator_update
    self.update_data_from_sensor()
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 181, in update_data_from_sensor
    (int(time_span_sensor._attr_native_value) == 0) and
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'unavailable'
```

and

```
Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:368
Integration: Sensori (documentation, issues)
First occurred: 18.29.09 (1 occurrences)
Last logged: 18.29.09

Error while setting up mobile_alerts platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 368, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1281, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/mobile_alerts/base.py", line 193, in async_added_to_hass
    self._handle_coordinator_update()
  File "/config/custom_components/mobile_alerts/base.py", line 220, in _handle_coordinator_update
    entity.update_data_from_sensor()
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 181, in update_data_from_sensor
    (int(time_span_sensor._attr_native_value) == 0) and
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'unavailable'
```

and 

```
Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:507
Integration: Sensori (documentation, issues)
First occurred: 18.29.09 (1 occurrences)
Last logged: 18.29.09

Error adding entities for domain sensor with platform mobile_alerts
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1281, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/mobile_alerts/base.py", line 193, in async_added_to_hass
    self._handle_coordinator_update()
  File "/config/custom_components/mobile_alerts/base.py", line 220, in _handle_coordinator_update
    entity.update_data_from_sensor()
  File "/config/custom_components/mobile_alerts/binary_sensor.py", line 181, in update_data_from_sensor
    (int(time_span_sensor._attr_native_value) == 0) and
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'unavailable'
```

**Comment by kucerazdenek** [comment]
@larobo and others, I would like to ask you, if you would be so kind and someone of you would be able to amend current repo and add 
state_class: measurement to all sensors to keep long term statistic in HA? All values are purged after 10 days currently, when I checked other temperature sensors provided by other integrations, those keep long term statistics and only difference in sensor state is state_class.

**This one (MFA) is purged**
![image](https://github.com/PlusPlus-ua/ha_mobilealerts/assets/121700009/55d0f896-a79e-4117-b376-0c6530e31e61)

**This one (Shelly) keeps long-term statistics:**, has state_class set
![image](https://github.com/PlusPlus-ua/ha_mobilealerts/assets/121700009/0118e525-7ca7-4a9e-be75-5f24eab071af)

**Comment by kucerazdenek** [comment]
I tried to do it by myself, just added single row of code
**state_class=SensorStateClass.MEASUREMENT,**
 into following block in sensor.py:

```
descriptions: dict[MeasurementType, SensorEntityDescription] = {
    MeasurementType.TEMPERATURE: SensorEntityDescription(
        key=None,
        device_class=SensorDeviceClass.TEMPERATURE,
        state_class=SensorStateClass.MEASUREMENT,
        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
```
but, does anyone know if only new readings pulled after this change will be kept for period longer than 10 days, or this change should be retrospective?
I am asking, because I did this change yesterday afternoon and data older than 10 days are still purged this morning.

Current status of temperature sensor in HA (state_class is present now)
![image](https://github.com/PlusPlus-ua/ha_mobilealerts/assets/121700009/7cab7ec9-9702-4db7-86a2-18a5a65af9c2)

**Comment by hiatsu0** [comment]
If it shows attributes like that, then long term stats should work indeed. But only for states that had that attribute I think. If you are impatient to wait 10 days, set your purge delay to 1 day and see if the stats remain :)

Great work btw, will add this line myself too!

**Comment by kucerazdenek** [comment]
Make sense, thanks for explanation! :)
I will probably do the same for humidity sensors.

**Comment by marek-veber** [comment]
I have been solving the same issue https://github.com/PlusPlus-ua/ha_mobilealerts/pull/25 :D

I had two issues:
1) the need of `dataclasses.replace`
2) after the battery stoped working - > then the value `unavailable` cannot be converted by `int(...)` of `float(....)`

I expect that the 1st issue will be also solved by [this](https://github.com/PlusPlus-ua/ha_mobilealerts/pull/24) PullRequest.  I'll check it on my instance.
2nd issue should be addressed by a separate new GitHub issue and PullRequest ;)

**Comment by berndh11** [comment]
thanks again to Iminet72! He solved perfectly the problem in the pull request.

**Comment by uschindler** [comment]
Hi,
I tried to install the fork of @Iminet72. It install successfully and no python errors are happening. It detects the MobileAlerts Gateway successfully and adds the config for it. But when it then starts the integration it hangs repeating the same over and over (the first part of the log is printed with debugging enabled for the integration, so it successfully gets all information from the gateway and in CAN READ the configuration). But at end it can't enable the integration.

```
2024-10-19 15:55:36.154 DEBUG (MainThread) [custom_components.mobile_alerts.config_flow] async_step_single_gateway gateway MOBILEALERTS-Gateway V1.50, SerialNo: xxxx (id: xxxx)
Use DHCP: Yes
DHCP IP: 192.168.1.7
Fixed IP: 192.168.1.222
Fixed Netmask: 255.255.255.0
Fixed Gateway: 192.168.1.254
Fixed DNS: 192.168.1.253
Cloud Server: www.data199.com
Use Proxy: No
Proxy Server: 192.168.1.1
Proxy Port: 8080
Send data to cloud: Yes
Last Contact: Sat Oct 19 15:55:34 2024
2024-10-19 15:55:38.499 DEBUG (MainThread) [custom_components.mobile_alerts.config_flow] async_step_single_gateway gateway MOBILEALERTS-Gateway V1.50, SerialNo: xxxx(id: xxxxxx)
Use DHCP: Yes
DHCP IP: 192.168.1.7
Fixed IP: 192.168.1.222
Fixed Netmask: 255.255.255.0
Fixed Gateway: 192.168.1.254
Fixed DNS: 192.168.1.253
Cloud Server: www.data199.com
Use Proxy: No
Proxy Server: 192.168.1.1
Proxy Port: 8080
Send data to cloud: Yes
Last Contact: Sat Oct 19 15:55:34 2024
2024-10-19 15:55:38.500 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
2024-10-19 15:56:13.863 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
2024-10-19 15:56:54.379 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
2024-10-19 15:57:44.668 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
2024-10-19 15:58:55.096 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
2024-10-19 16:00:45.376 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
2024-10-19 16:02:35.572 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
```

This repeats forever.

Unfortunately the Exception as printed in the UI (interestingly you can't find it in the log file its only printed on the integration configuration page) is not very helpful, it only says that it can't initialize the gateway. It is thrown here: https://github.com/Iminet72/ha_mobilealerts/blob/1d5a70b7f7da7e831ab3139ee82cd9adf430f1ef/custom_components/mobile_alerts/__init__.py#L30-L33

It returns false here in the low level gateway/proxy implementation causing the above exception: https://github.com/PlusPlus-ua/python-mobilealerts/blob/80b4591a3998dcbe62266e6f1294104b4ddeffd5/mobilealerts/gateway.py#L91-L100

I don't understand why the discovery of the gateway worked, but when it wants to enable the integration for the given gateway it fails while parsing the configuration.

Any idea what goes wrong? The above log messages tell me that it find the gateway and can read its configiuration, but why does it fail later when it sets it up for production? I am using Hass 2024.8.3.

**Comment by newslettersns2000** [comment]
I have no Problems

**Comment by uschindler** [comment]
> Hi, I tried to install the fork of @Iminet72. It install successfully and no python errors are happening. It detects the MobileAlerts Gateway successfully and adds the config for it. But when it then starts the integration it hangs repeating the same over and over (the first part of the log is printed with debugging enabled for the integration, so it successfully gets all information from the gateway and in CAN READ the configuration). But at end it can't enable the integration.
> 
> ```
> 2024-10-19 15:55:36.154 DEBUG (MainThread) [custom_components.mobile_alerts.config_flow] async_step_single_gateway gateway MOBILEALERTS-Gateway V1.50, SerialNo: xxxx (id: xxxx)
> Use DHCP: Yes
> DHCP IP: 192.168.1.7
> Fixed IP: 192.168.1.222
> Fixed Netmask: 255.255.255.0
> Fixed Gateway: 192.168.1.254
> Fixed DNS: 192.168.1.253
> Cloud Server: www.data199.com
> Use Proxy: No
> Proxy Server: 192.168.1.1
> Proxy Port: 8080
> Send data to cloud: Yes
> Last Contact: Sat Oct 19 15:55:34 2024
> 2024-10-19 15:55:38.499 DEBUG (MainThread) [custom_components.mobile_alerts.config_flow] async_step_single_gateway gateway MOBILEALERTS-Gateway V1.50, SerialNo: xxxx(id: xxxxxx)
> Use DHCP: Yes
> DHCP IP: 192.168.1.7
> Fixed IP: 192.168.1.222
> Fixed Netmask: 255.255.255.0
> Fixed Gateway: 192.168.1.254
> Fixed DNS: 192.168.1.253
> Cloud Server: www.data199.com
> Use Proxy: No
> Proxy Server: 192.168.1.1
> Proxy Port: 8080
> Send data to cloud: Yes
> Last Contact: Sat Oct 19 15:55:34 2024
> 2024-10-19 15:55:38.500 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> 2024-10-19 15:56:13.863 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> 2024-10-19 15:56:54.379 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> 2024-10-19 15:57:44.668 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> 2024-10-19 15:58:55.096 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> 2024-10-19 16:00:45.376 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> 2024-10-19 16:02:35.572 DEBUG (MainThread) [custom_components.mobile_alerts] async_setup_entry {'created_at': '2024-10-19T13:55:38.499783+00:00', 'data': {'send_data_to_cloud': True}, 'disabled_by': None, 'domain': 'mobile_alerts', 'entry_id': '01JAJGZWC3CH0DMQ1ZKY7SW2T2', 'minor_version': 1, 'modified_at': '2024-10-19T13:55:38.499791+00:00', 'options': {}, 'pref_disable_new_entities': False, 'pref_disable_polling': False, 'source': 'user', 'title': 'xxxx', 'unique_id': 'xxxx', 'version': 1}
> ```
> 
> This repeats forever.
> 
> Unfortunately the Exception as printed in the UI (interestingly you can't find it in the log file its only printed on the integration configuration page) is not very helpful, it only says that it can't initialize the gateway. It is thrown here: https://github.com/Iminet72/ha_mobilealerts/blob/1d5a70b7f7da7e831ab3139ee82cd9adf430f1ef/custom_components/mobile_alerts/__init__.py#L30-L33
> 
> It returns false here in the low level gateway/proxy implementation causing the above exception: https://github.com/PlusPlus-ua/python-mobilealerts/blob/80b4591a3998dcbe62266e6f1294104b4ddeffd5/mobilealerts/gateway.py#L91-L100
> 
> I don't understand why the discovery of the gateway worked, but when it wants to enable the integration for the given gateway it fails while parsing the configuration.
> 
> Any idea what goes wrong? The above log messages tell me that it find the gateway and can read its configiuration, but why does it fail later when it sets it up for production? I am using Hass 2024.8.3.

I found out where this issue comes from. It has nothing to do with the changes proposed here. The issue is simple:
My Hass server has multiple network interfaces and IP addresses. When the config workflow does the lookuop it passes HASS IP as configured in the network interfaces as "base" address for the discovery. This IP address is retrieved from hass, see here:

https://github.com/PlusPlus-ua/ha_mobilealerts/blob/8143b1107d8c7ca93d8a19d56397b28cc3ec33c8/custom_components/mobile_alerts/config_flow.py#L115-L120

Unfortunately when the implementation starts it can't find the gateway and more (to set the proxy port and IP address) of proxy, because the broadcast is sent with None as source address. In case of multiple network interfaces, the system does not know what interface to use to send the broadcast for reconfiguring the gateway.

Basically, my quick hack was simple, just modify the __main__.py file and hardcode my internal IP address as 2nd parameter to `new Gateway()`. The correct fix is to move the proxy_ip a few lines below up:
```
    proxy_ip = await async_get_source_ip(hass, gateway_ip)
```
and use it for discovery.

I will open a separate issue. As this repository seems dead, is anybody willing to host a patched/forked version? @Iminet72 are you willing to take over the repo, i would open a PR in your fork then? Otherwise I can also fork the repo and make my own impl with all fixes applied (including the one with broadcast source address).

**Comment by uschindler** [comment]
I opened #26 about the broadcast source IP address issue.

**Comment by berndh11** [comment]
Hallo Uwe,
ich gehe davon aus dass du Deutsch sprichst nach deinem Namen zu urteilen.
Am besten ich schicke dir mal die beiden Dateien per E-Mail und dann kannst
du die einfach in Home Assistant hochladen. Dann müsste alles gehen. Wenn
es dann immer noch nicht geht, ist mit deiner Konfiguration etwas nicht in
Ordnung bzw irgendein Update macht Schwierigkeiten. Ich habe allerdings
alle Updates installiert und bei mir läuft alles tadellos.

Viele Grüße.

Bernd

Uwe Schindler ***@***.***> schrieb am So., 20. Okt. 2024,
00:41:

**Comment by uschindler** [comment]
Hi Bernd,

> Hallo Uwe, ich gehe davon aus dass du Deutsch sprichst nach deinem Namen zu urteilen. Am besten ich schicke dir mal die beiden Dateien per E-Mail und dann kannst du die einfach in Home Assistant hochladen. Dann müsste alles gehen. Wenn es dann immer noch nicht geht, ist mit deiner Konfiguration etwas nicht in Ordnung bzw irgendein Update macht Schwierigkeiten. Ich habe allerdings alle Updates installiert und bei mir läuft alles tadellos. Viele Grüße. Bernd 

Das Problem mit der Discovery hat mit dem Problem hier nichts zu tun, ich habe neuen Issue aufgemacht. Bitte vollständig lesen. Siehe #26. Es handelt sich um einen weiteren Fehler der in Konfigurationen mit mehreren Netzwerkenschnittstellen passiert. Es wurde nur vergessen die spezifische/passende Source IP Adresse bei Aktivierung des Gateways zu benutzen, was dann dazu führt dass das Gateway später nicht mehr gefunden wird.

Uwe
